jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -174,9 +174,10 @@ def _on_viewer_data_changed(self, msg):
             return
 
         viewer = self.app.get_viewer(self._default_spectrum_viewer_reference_name)
-        statistic = getattr(viewer, 'function', None)
-        viewer_data = self.app._jdaviz_helper.get_data(data_label=msg.data.label,
-                                                       statistic=statistic)
+        get_data_kwargs = {'data_label': msg.data.label}
+        if self.config == 'cubeviz':
+            get_data_kwargs['function'] = getattr(viewer, 'function', None)
+        viewer_data = self.app._jdaviz_helper.get_data(**get_data_kwargs)
```

Comment:
While testing this, I ran into a problem further down this file (line 404) where `get_data` is called without data_label set with multiple data present. This can be reproduced in the CubevizExample notebook by making a spectral region (Subset 1) in the specviz viewer followed by a continuum (Subset 2) in the same viewer. Would it be possible to add that fix to this PR or should I make a separate issue?

Reply:
I added passing `self.dataset.selected`... please let me know if it fixes the case.  Since this has not been released yet, I think putting it in the same PR is fine.